qdglue/metrics/grid_based_metrics.py

```diff
@@ -0,0 +1,290 @@
+import dataclasses
+from abc import ABC, abstractmethod
+from typing import Dict, Tuple
+
+import gymnasium
+import numpy as np
+
+from qdglue.types import Feature, Fitness
+
+
+@dataclasses.dataclass
+class DiscreteArchiveMetrics:
```

Comment:
I haven't had to time to think this through, but I think we should generally aim to make this class "self-contained" -- i.e. if all I get is this class, I shouldn't have to look back at my QD algorithm to figure out what these metrics mean. This way, I can easily pass this object around and someone else can interpret what the metrics mean.

Reply:
As said above, I was hesitating about that. And I do not have any strong opinion.

I just added a `discrete_archive_metrics_functor` attribute to the metrics dataclass, so that we can find all the info about the metrics calculator directly in the metrics object.

Reply:
See other comment below.